### PR TITLE
[OPS-1636] Redirect-Config.json file for Changes in OPS Land to redirect state/01 --> state/alabama via 301

### PR DESCRIPTION
**Pull request by MaxwellKendall**
**High level description:**
OPS will consume this file to redirect our state URLS at the network level incase someone goes directly to `/state/01`. [New deploy script](https://github.com/fedspendingtransparency/usaspending-config/pull/1001) will generate and consume this file.

**Technical details:**
- build script to generate .json
- build config updated to expect .json file to exist in root directory, and copy into build artifact `/public` dir
- new npm script for deploy script in usaspending-config to use

**JIRA Ticket:**
[OPS-1636](https://federal-spending-transparency.atlassian.net/browse/OPS-1636)

The following are ALL required for the PR to be merged:

Author:
- [x] Linked to this PR in JIRA ticket
`N/A` Scheduled demo including Design/Testing/Front-end OR Provided instructions for testing in JIRA and PR `if applicable`
`N/A` Verified cross-browser compatibility: Chrome, Safari, Firefox, Internet Explorer, Edge
`N/A` Verified mobile/tablet/desktop/monitor responsiveness
`N/A` Verified that this PR does not create any *new* accessibility issues (via Axe Chrome extension)
`N/A` Added Unit Tests for helper functions, reducers, models and Container/Component Interactivity Expectations `if applicable` [React Testing Library](react-testing-library.md)
`N/A`  [API contract](https://github.com/fedspendingtransparency/usaspending-api/tree/dev/usaspending_api/api_contracts) updated `if applicable`
`N/A` [Component Library Integration Status Report](https://github.com/fedspendingtransparency/data-act-documentation/blob/data-transparency-ui/frontend_apps/component-library-integration-status.md) updated `if applicable`

Reviewer(s):
`N/A`  Design review complete `if applicable`
`N/A`  [API #1234](https://github.com/fedspendingtransparency/usaspending-api/pull/1234) merged concurrently `if applicable`
- [x] Code review complete
- [x] OPS code review complete @bminahan-kc 
- [x] Tested in sandbox w/ lambdas

### CODE REVIEW

**Review by bminahan-kc — APPROVED** [review]
👍

File: scripts/buildRedirectConfig.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import fs from 'fs';
+import path from 'path';
+import { stateNameByFipsId } from '../src/js/dataMapping/state/stateNames';
+import { URLifyStateName } from '../src/js/helpers/stateHelper';
+
+const legacyRedirects = {
+    "/^Pages/Default.aspx/": "/#",
```
Comment (bminahan-kc) [inline scripts/buildRedirectConfig.js:7]: ^/Pages/Default.aspx

File: scripts/buildRedirectConfig.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    "/^Pages/AdvancedSearch.aspx/": "/search",
```
Comment (bminahan-kc) [inline scripts/buildRedirectConfig.js:8]: ^/Pages/AdvancedSearch.aspx

File: scripts/buildRedirectConfig.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    "/^DownloadCenter/Pages/DataDownload.aspx/": "/download_center/custom_award_data",
```
Comment (bminahan-kc) [inline scripts/buildRedirectConfig.js:9]: ^/DownloadCenter/Pages/DataDownload.aspx

File: scripts/buildRedirectConfig.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    "/^DownloadCenter/Pages/dataarchives.aspx/": "/download_center/award_data_archive",
```
Comment (bminahan-kc) [inline scripts/buildRedirectConfig.js:10]: ^/DownloadCenter/Pages/dataarchives.aspx

File: scripts/buildRedirectConfig.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    "/^covid-19/": "/disaster/covid-19"
```
Comment (bminahan-kc) [inline scripts/buildRedirectConfig.js:11]: ^/covid-19

**Review by bminahan-kc — CHANGES_REQUESTED** [review]
looks like state redirects are fine but the legacy ones also need that change

**Review by bminahan-kc — APPROVED** [review]

**Review by ebdabbs — APPROVED** [review]

**Pull request merged**